Add support to connect to HPE 1920 and 1950 OfficeConnect switches

spamik opened the pull request:
HPE 1920 and 1950 switches has standard Comware5 / Comware7 CLI but it needs to be enabled with special command sent to the CLI. This patch add new hpe_officeconnect_model argument to HPComwareBase class. When this argument is populated with hpe1920 or hpe1950 string value, it send magic command immediately after establishing connection with switch and therefore these switches can be handled as standard Comware switches.

Commit:
Add support to connect to HPE 1920 and 1950 OfficeConnect switches

HPE 1920 and 1950 switches has standard Comware5 / Comware7 CLI but it needs to be enabled with special command sent to the CLI. This patch add new hpe_officeconnect_model argument to HPComwareBase class. When this argument is populated with hpe1920 or hpe1950 string value, it send magic command immediately after establishing connection with switch and therefore these switches can be handled as standard Comware switches.

evilmonkey19 commented:
I can confirm that this is a [special case](https://networkguy.de/enabling-advanced-cli-on-hp-v1910-switches/) those switches :+1:

evilmonkey19 commented:
In this case instead of adding some argument to the credentials i would suggest creating a new device type or issue the command version on startup and based on that decide. If you go for the second option I can help you creating the template in [NTC-Templates](https://github.com/networktocode/ntc-templates)

spamik commented:
@evilmonkey19 issuing display version command was my first attempt to keep it simple without configuration. But problem is that even display version is not possible without enabling CLI :-/ In basic version there were some commands from which output is possible to determine what type switch is. But I've ended on that problem that these commands has long output which not fit on one screen (----more---- placer is returned to CLI). Command which disable paging is also available only after enabling full CLI. And even pipe (something like "| include 1920") is also available only with full CLI. So I didn't find the way how to detect switch model before enabling full CLI with something which not lead to terminal paging issue :-/ (If you have some idea, I can test it, i have bunch of both types of these switches).

evilmonkey19 commented:
Then the best solution is to create a separated device_type for these type of switches and there is no way to detect them beforehand.

Commit:
Revert "Add support to connect to HPE 1920 and 1950 OfficeConnect switches"

This reverts commit 4418d18be9186fd7fd0adc62e4c7a70996dc222d.

Commit:
Add support to connect to HPE 1920 and 1950 OfficeConnect switches

HPE 1920 and 1950 switches has standard Comware5 / Comware7 CLI but it needs to be enabled with special command sent to the CLI. This patch add new device communication classes based on current HPComwareBase class. These drivers during session initializations send special commands to enable full CLI and therefore after this initialization is possible to work with them like with regular Comware switches. For using these drivers were added 2 new device types: hp_comware1920 and hp_comware1950.

spamik commented:
OK, I've reverted previous commit and I've added 2 new device types - one for 1920 series, second for 1950 series.

evilmonkey19 commented:
Awesome! IMHO, It does look much cleaner not the codebase but also the usability for the user. 😃

Soullego commented:
@spamik i would suggest also add 1910 model.

spamik commented:
@Soullego Do you have some info what steps are needed on 1910 to enable full cli? I unfortunately didn't have any 1910 on stock to try it.

Soullego commented:
@spamik yeah, same as 1920 but different password
```
_cmdline-mode on
y
512900
```

Commit:
Add support to connect to HPE 1910 OfficeConnect switches

Added classes for HPE OfficeConnect 1910 switches - same behvaiour like for HPE 1920, just with different password to enable full CLI mode

spamik commented:
I've added new class for HPE 1910, so if someone can test it...

ktbyers commented:
I am going to close this one as I don't really know what to do about the hard-coded backdoor passwords (and everytime I look at it, I basically defer/punt it).
